tests/test_official.py

```diff
@@ -264,9 +287,222 @@ def check_defaults_type(ptb_param: inspect.Parameter) -> bool:
+
+        assert mapped_type is Sequence
+        # This means it is Array of Array of [obj]
+        if "Array of Array of" in tg_param_type:
+            # print('isinstance 1', Sequence[Sequence[obj]], ptb_annotation)
+            return any(Sequence[Sequence[o]] == ptb_annotation for o in unionized_objs)
+
+        # This means it is Array of [obj]
+        # print('isinstance 2', mapped_type[obj], ptb_annotation)
+        return any(mapped_type[o] == ptb_annotation for o in unionized_objs)
+
+    DEFAULT_VAL_PARAMS = {
+        "parse_mode",
+        "disable_notification",
+        "allow_sending_without_reply",
+        "protect_content",
+        "disable_web_page_preview",
+        "explanation_parse_mode",
+    }
+
+    # Special case for when the parameter is a default value parameter
+    if ptb_param.name in DEFAULT_VAL_PARAMS:
+        # Check if it's DVInput or ODVInput
+        for param_type in [DVInput, ODVInput]:
+            parsed = param_type[mapped_type]
+            if ptb_annotation == parsed:
+                return True
+        # print(f"cause {mapped_type} != {ptb_annotation} for {ptb_param.name}")
+        return False
+
+    # Special case for send_* methods where we accept more types than the official API:
+    additional_types = {
+        "photo": ForwardRef("PhotoSize"),
+        "video": ForwardRef("Video"),
+        "video_note": ForwardRef("VideoNote"),
+        "audio": ForwardRef("Audio"),
+        "document": ForwardRef("Document"),
+        "animation": ForwardRef("Animation"),
+        "voice": ForwardRef("Voice"),
+        "sticker": ForwardRef("Sticker"),
+    }
+
+    if (
+        ptb_param.name in additional_types
+        and not isinstance(mapped_type, tuple)
+        and obj.__name__.startswith("send")
+    ):
+        print("additional_types ")
+        mapped_type = mapped_type | additional_types[ptb_param.name]
+
+    # Special cases for other methods that accept more types than the official API, and are
+    # too complex to compare/predict with official API:
+    exceptions = {  # (param_name, is_class): reduced form of annotation
+        ("correct_option_id", False): int,  # actual: Literal
+        ("file_id", False): str,  # actual: Union[str, objs_with_file_id_attr]
+        ("invite_link", False): str,  # actual: Union[str, ChatInviteLink]
+        ("provider_data", False): str,  # actual: Union[str, obj]
+        ("callback_data", True): str,  # actual: Union[str, obj]
+        ("media", True): str,  # actual: Union[str, InputMedia*, FileInput]
+        ("data", True): str,  # actual: Union[IdDocumentData, PersonalDetails, ResidentialAddress]
+    }
+
+    for (param_name, expected_class), exception_type in exceptions.items():
+        if ptb_param.name == param_name and is_class is expected_class:
+            ptb_annotation = exception_type
+
+    # Special case for datetimes
+    if re.search(r"([_]+|\b)date[^\w]*\b", ptb_param.name):
```

Comment:
How about instead searching for "unix time" in the tg dowstring?

Reply:
a quick ctrl+f tells me it's possible to do that as well. I think I'll add an `or` here just to be more careful